crates/pink/sidevm/examples/recv_messages/src/lib.rs

```diff
@@ -0,0 +1,21 @@
+use log::info;
+
+use pink_sidevm as sidevm;
+use sidevm::{logger::Logger, ocall};
+
+#[sidevm::main]
+async fn main() {
+    Logger::with_max_level(log::Level::Trace).init();
+    ocall::enable_ocall_trace(true).unwrap();
+
+    loop {
+        info!("waiting next message...");
+        if let Some(message) = sidevm::channel::input_messages().next().await {
+            let text_message = String::from_utf8_lossy(&message);
+            info!("received message: {}", text_message);
+        } else {
+            info!("no message received");
```

Comment:
The tx-end is in the host-runtime, so ink contract unable to close this channel's tx end. It can close the rx but access the "fd" after close is a typically UAF.
When we kill a instance, the tx in the runtime is droped while the instance is running. At this time the rx-end might receive a None which indicate the channel has been closed.

Reply:
Would be nice to add the comment to the code

Reply:
Added document to the `fn input_messages`